### PR TITLE
chore: use @guardian eslint configs

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- counterpart to #317 (and dependant on it)
- uses global guardian configs

## Why?

- consistency and good practise

### CODE REVIEW

File: cypress/integration/sourcepoint-tcfv2.spec.js [hunk 1]
```diff
@@ -57,7 +55,7 @@ skipOn(Cypress.env('CI') === 'true', () => {
 				.find(`button[title="${buttonTitle}"]`)
 				.click();
 
-			// eslint-disable-next-line cypress/no-unnecessary-waiting
+			// eslint-disable-next-line cypress/no-unnecessary-waiting -- should we do this?
```
Comment (mxdvl) [inline cypress/integration/sourcepoint-tcfv2.spec.js:58]: I think we can wait for a cookie to be present… let’s look at how it’s done in CCPA and raise another PR.

File: src/aus/sourcepoint.test.js [hunk 2]
```diff
@@ -44,7 +43,7 @@ describe('Sourcepoint CCPA', () => {
 	it('points at a real file', (done) => {
 		init();
 		const src = document
-			?.getElementById('sourcepoint-aus-lib')
+			.getElementById('sourcepoint-aus-lib')
```
Comment (mxdvl) [inline src/aus/sourcepoint.test.js:46]: Why is this gone?

File: src/lib/domain.ts [hunk 3]
```diff
@@ -2,6 +2,6 @@ let isGuardian: boolean | undefined;
 
 export const isGuardianDomain = (): boolean => {
 	if (typeof isGuardian === 'undefined')
-		isGuardian = /\.theguardian\.com$/.test(window.location.host);
+		isGuardian = window.location.host.endsWith('.theguardian.com');
```
Comment (mxdvl) [inline src/lib/domain.ts:5]: ’tis such Poetry…

File: src/onConsentChange.ts [hunk 4]
```diff
@@ -51,9 +49,7 @@ export const onConsentChange: (fn: Callback) => void = (callBack) => {
 	// if consentState is already available, invoke callback immediately
 	getConsentState()
 		.then((consentState) => {
-			if (consentState) {
-				invokeCallback(newCallback, consentState);
-			}
+			invokeCallback(newCallback, consentState);
```
Comment (mxdvl) [inline src/onConsentChange.ts:52]: So `consentState` is always defined?

File: src/types/tcfv2/TCData.ts [hunk 5]
```diff
@@ -88,15 +88,6 @@ export interface TCData {
 			consents: TCFv2ConsentList;
 			legitimateInterests: TCFv2ConsentList;
 		};
-		restrictions: {
-			[key: string]: {
-				/**
-				 * 0 - Not Allowed
-				 * 1 - Require Consents
-				 * 2 - Require Legitimate Interest
-				 */
```
Comment (mxdvl) [inline src/types/tcfv2/TCData.ts:97]: We’ve lost this comment! Worth turning into a type?

File: tsconfig.json [hunk 6]
```diff
@@ -10,5 +10,5 @@
     "strict": true,
     "target": "ES2020"
   },
-  "exclude": ["node_modules", "dist", "coverage", "cypress", ".*"]
+  "include": ["*.js", ".*.js", "src"]
```
Comment (mxdvl) [inline tsconfig.json:13]: What about `node_modules`?

**Review by mxdvl — APPROVED** [review]
Few comments, but generally 👍

File: tsconfig.json (hunk 6 above)
Comment (sndrs) [reply tsconfig.json:13]: it is excluded by default

File: src/lib/domain.ts (hunk 3 above)
Comment (sndrs) [reply src/lib/domain.ts:5]: yeah auto-fixed too!

File: src/types/tcfv2/TCData.ts (hunk 5 above)
Comment (sndrs) [reply src/types/tcfv2/TCData.ts:97]: oh no that was an oversight, restoring it

File: src/aus/sourcepoint.test.js (hunk 2 above)
Comment (sndrs) [reply src/aus/sourcepoint.test.js:46]: i guess it was not necesary, `document` must be defined in jest types. it was removed by a fixable rule

File: src/onConsentChange.ts (hunk 4 above)
Comment (sndrs) [reply src/onConsentChange.ts:52]: that's what typescript thinks. i wonder if we should update the type so it can be `undefined`?

Comment (mxdvl) [reply src/onConsentChange.ts:52]: In TS we trust.
